httpie/output/formatters/colors.py

```diff
@@ -237,3 +268,142 @@ class Solarized256Style(pygments.style.Style):
         pygments.token.Token: BASE1,
         pygments.token.Token.Other: ORANGE,
     }
+
+
+def format_style(raw_styles, shade):
+    def get_shade(part):
+        if part not in COLOR_PALETTE:
+            return part
+
+        color_code = COLOR_PALETTE[part]
+        if isinstance(color_code, dict) and shade in color_code:
+            return color_code[shade]
+        else:
+            return color_code
+
+    def format_value(value):
+        return " ".join(
+            get_shade(part)
+            for part in value.split()
+        )
+
+    return {
+        key: format_value(value)
+        for key, value in raw_styles.items()
+    }
+
+
+def make_styles(name, raw_styles):
+    for mode, shade in [
+        ('light', '700'),
+        ('universal', '600'),
+        ('dark', '500')
+    ]:
+        yield type(
+            name.format(mode=mode.title()),
+            (pygments.style.Style,),
+            {
+                'styles': format_style(raw_styles, shade)
+            }
+        )
```

Comment:
These are tied to the Pie themes, so I’d either reflect that in their names or move them to `palette`.